Sources/Error Handling/SKError+Extensions.swift

```diff
@@ -17,49 +17,51 @@ import StoreKit
 extension SKError {
 
     // swiftlint:disable:next cyclomatic_complexity
-    func toPurchasesErrorCode() -> ErrorCode {
+    func toPurchasesError() -> Error {
         switch self.code {
         case .cloudServiceNetworkConnectionFailed,
              .cloudServiceRevoked,
              .overlayTimeout,
              .overlayPresentedInBackgroundScene:
-            return .storeProblemError
+            return ErrorUtils.storeProblemError(error: self)
         case .clientInvalid,
              .paymentNotAllowed,
              .cloudServicePermissionDenied,
              .privacyAcknowledgementRequired:
-            return .purchaseNotAllowedError
+            return ErrorUtils.purchaseNotAllowedError(error: self)
         case .paymentCancelled,
              .overlayCancelled:
-            return .purchaseCancelledError
+            return ErrorUtils.purchaseCancelledError(error: self)
         case .paymentInvalid,
-             .unauthorizedRequestData,
-             .missingOfferParams,
-             .invalidOfferPrice,
-             .invalidSignature,
-             .invalidOfferIdentifier:
-            return .purchaseInvalidError
+            .unauthorizedRequestData:
+            return ErrorCode.purchaseInvalidError
         case .storeProductNotAvailable:
-            return .productNotAvailableForPurchaseError
-        case .ineligibleForOffer,
-             .overlayInvalidConfiguration,
+            return ErrorUtils.productNotAvailableForPurchaseError(error: self)
+        case .overlayInvalidConfiguration,
              .unsupportedPlatform:
-            return .purchaseNotAllowedError
+            return ErrorUtils.purchaseNotAllowedError(error: self)
+        case .ineligibleForOffer:
+            return ErrorUtils.ineligibleError(error: self)
+        case .missingOfferParams,
+            .invalidOfferPrice,
+            .invalidSignature,
+            .invalidOfferIdentifier:
```

Comment:
This were being marked as `.purchaseInvalidError` but we have a more specific error.